### PR TITLE
[8/N] Fixes clang-tidy warnings in c10/{core,util}/*.h

### PR DESCRIPTION
**Pull request by cyyever**
This patch enables clang-tidy coverage on c10/**/*.h and contains other fixes.

### CODE REVIEW

File: c10/util/string_view.h [hunk 1]
```diff
@@ -488,7 +488,7 @@ class basic_string_view final {
       const noexcept {
     if (pos + 1 <= size()) {
       for (size_type cur = pos; cur < size(); ++cur) {
-        if (condition(at_(cur))) {
+        if (std::forward<Condition>(condition)(at_(cur))) {
```
Comment (Skylion007) [inline c10/util/string_view.h:491]: This smells like a false positive? How can something be forwarded again, and again in a loop? Wouldn't that cause a use-after-move bug?

**Review by Skylion007 — CHANGES_REQUESTED** [review]
One major issue smells like a false positive.

File: c10/util/string_view.h (hunk 1 above)
Comment (cyyever) [reply c10/util/string_view.h:491]: Oh, it should be a false positive

**Comment by cyyever** [comment]
@Skylion007 Fixed

File: c10/util/string_view.h [hunk 2]
```diff
@@ -497,6 +498,7 @@ class basic_string_view final {
   }
 
   template <class Condition>
+  // NOLINTNEXTLINE(cppcoreguidelines-missing-std-forward)
   constexpr size_type find_last_if_(size_type pos, Condition&& condition)
```
Comment (Skylion007) [inline c10/util/string_view.h:502]: Why is this a universal forward reference in the first place?

File: .lintrunner.toml [hunk 3]
```diff
@@ -242,6 +242,7 @@ code = 'CLANGTIDY'
 include_patterns = [
     'aten/src/ATen/core/*.cpp',
     'c10/**/*.cpp',
+    'c10/core/**/*.h',
```
Comment (Skylion007) [inline .lintrunner.toml:245]:
```suggestion
    'c10/core/**/*.h*',
```
in case someone ever puts hpp files in there.

Comment (cyyever) [reply .lintrunner.toml:245]: No hpp in PyTorch

**Review by Skylion007 — APPROVED** [review]

File: .lintrunner.toml (hunk 3 above)
Comment (cyyever) [reply .lintrunner.toml:245]: There may be someone using hpp in the future. I will change the pattern in the next patch

**Pull request closed without merging**